arch/src/x86_64/regs.rs

```diff
@@ -94,15 +94,18 @@ pub fn setup_msrs(vcpu: &VcpuFd) -> Result<()> {
 ///
 /// * `vcpu` - Structure for the VCPU that holds the VCPU's fd.
 /// * `boot_ip` - Starting instruction pointer.
-/// * `boot_sp` - Starting stack pointer.
-/// * `boot_si` - Must point to zero page address per Linux ABI.
-pub fn setup_regs(vcpu: &VcpuFd, boot_ip: u64, boot_sp: u64, boot_si: u64) -> Result<()> {
```

Comment:
Maybe there's value in keeping all the method parameters, just in case someone would like to use different values for some reason, or we want to move away from the hardcoded layout at some point. This is actually very similar to your other PR where you have the per arch `IRQ_BASE` and `IRQ_MAX`. The `device_manager::new()` method got a new parameter (the `(base, max)` pair), instead of taking those values directly from `arch::layout::IRQ_BASE` and `arch::layout::IRQ_MAX`. I think the more flexible approach is preferable just because it's more future-proof. Happy to chat about this.

Reply:
I did take into account also the advantages of keeping them that way. We can talk about it offline.